lib/daru/vector.rb

```diff
@@ -170,21 +171,26 @@ def map!(&block)
     #   vecarr = Daru::Vector.new [1,2,3,4], index: [:a, :e, :i, :o]
     #   vechsh = Daru::Vector.new({a: 1, e: 2, i: 3, o: 4})
     def initialize source, opts={}
-      index, source = parse_source(source, opts)
-      set_name opts[:name]
+      if opts[:type] == :category
+        extend Daru::Category
```

Comment:
OK, let's leave it this way for now. For me, it looks like serious code smell curently, but in fact we can refactor it later as many times as we want to :)